src/FasterDBStore/Store.cs

```diff
@@ -0,0 +1,186 @@
+            public void SingleWriter(ref BufferKey key, ref BufferValue src, ref BufferValue dst) => dst = src;
+            public void ConcurrentReader(ref BufferKey key, ref Input input, ref BufferValue value, ref Output dst) => dst.Value = value;
+            public bool ConcurrentWriter(ref BufferKey key, ref BufferValue src, ref BufferValue dst) { dst = src; return true; }
+
+            public void ReadCompletionCallback(ref BufferKey key, ref Input input, ref Output output, Empty ctx, Status status) { }
+            public void UpsertCompletionCallback(ref BufferKey key, ref BufferValue value, Empty ctx) { }
+            public void RMWCompletionCallback(ref BufferKey key, ref Input input, Empty ctx, Status status) { }
+            public void DeleteCompletionCallback(ref BufferKey key, Empty ctx) { }
+            public void CheckpointCompletionCallback(Guid sessionId, long serialNum) { }
+        }
+
+        private readonly string StorePath;
+        private readonly IDevice log, objlog;
+        internal readonly FasterKV<BufferKey, BufferValue, Input, Output, Empty, FasterFunctions> db;
+
+        public Store(string path)
+        {
+            StorePath = path = Path.GetFullPath(path);
+
+            if (!Directory.Exists(path))
+            {
+                Directory.CreateDirectory(path);
+            }
+
+            log = Devices.CreateLogDevice(Path.Combine(path, "hlog.log"), preallocateFile: false, deleteOnClose: false, recoverDevice: true);
+            objlog = Devices.CreateLogDevice(Path.Combine(path, "hlog.obj.log"), preallocateFile: false, deleteOnClose: false, recoverDevice: true);
+
+            db = new FasterKV<BufferKey, BufferValue, Input, Output, Empty, FasterFunctions>
+                (1L << 20 /* TODO: I don't know a good value for this */, new FasterFunctions(),
+                new LogSettings { LogDevice = log, ObjectLogDevice = objlog },
+                new CheckpointSettings
+                {
+                    CheckpointDir = Path.Combine(path, "Snapshot"),
+                    CheckPointType = CheckpointType.Snapshot
+                },
+                new SerializerSettings<BufferKey, BufferValue>
+                {
+                    keySerializer = () => new BufferKeySerializer(),
+                    valueSerializer = () => new BufferValueSerializer()
+                }
+                );
+
+            // Each thread calls StartSession to register itself with FASTER
+
+            var uid = db.StartSession();
+
+            if (File.Exists(Path.Combine(path, "session.id")))
+            {
+                var data = new byte[16];
+                using (var file = File.OpenRead(Path.Combine(path, "session.id")))
+                {
+                    if (file.Read(data, 0, 16) == 16)
+                    {
+                        // Recover the last storage state
+
+                        db.Recover(new Guid(data));
```

Comment:
This line may throw the FasterException `Invalid index commit metadata for ID ***`, which is caused by the absence of the `completed.dat` file in the corresponding guid directory (under the snapshot directory).

Reply:
Maybe we can use try-catch to deal with the exception caused by the file missing issue.